spring-cloud-azure-keyvault-config/src/main/java/com/microsoft/azure/spring/cloud/keyvault/config/auth/AadKeyVaultCredentials.java

```diff
@@ -7,42 +7,37 @@
 package com.microsoft.azure.spring.cloud.keyvault.config.auth;
 
 import com.microsoft.aad.adal4j.AuthenticationContext;
-import com.microsoft.aad.adal4j.AuthenticationResult;
 import com.microsoft.azure.keyvault.authentication.KeyVaultCredentials;
-import lombok.AllArgsConstructor;
 
-import java.util.concurrent.ExecutorService;
-import java.util.concurrent.Executors;
-import java.util.concurrent.TimeUnit;
+import java.net.MalformedURLException;
+import java.util.concurrent.*;
 
 /**
  * Subclass of {@link KeyVaultCredentials}.
  */
-@AllArgsConstructor
 public class AadKeyVaultCredentials extends KeyVaultCredentials {
     // TODO: make it configurable.
     private static final int TIMEOUT_IN_SECONDS = 15;
+    private static final String AUTH_FAILED = "Failed to authenticate with Azure Key Vault.";
+
+    private final ExecutorService service = Executors.newSingleThreadExecutor();

```

Comment:
If we have many instances of this class, maybe static will be enough.

Reply:
Right. Let's leave it for now and optimize it later when it is really necessary.